src/Atc.Rest.Client/Builder/MessageRequestBuilder.cs

```diff
@@ -70,20 +77,43 @@ public IMessageRequestBuilder WithPathParameter<T>(string name, T value)
             where T : struct
             => WithPathParameter(name, value.ToString());
 
-        public IMessageRequestBuilder WithQueryParameter(string name, string? value)
+        public IMessageRequestBuilder WithHeaderParameter(string name, string value)
         {
-            if (!string.IsNullOrEmpty(value))
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                throw new ArgumentException($"'{nameof(name)}' cannot be null or whitespace", nameof(name));
+            }
+
+            if (string.IsNullOrWhiteSpace(value))
             {
-                // BANG added because nullable analyzer doesn't understand IsNullOrEmpty in .netstandard2.0
-                queryMapper[name] = value!;
+                throw new ArgumentException($"'{nameof(value)}' cannot be null or whitespace", nameof(value));
             }
 
+            headerMapper[name] = value;
+
             return this;
         }
 
-        public IMessageRequestBuilder WithQueryParameter<T>(string name, T value)
+        public IMessageRequestBuilder WithHeaderParameter<T>(string name, T value)
             where T : struct
-            => WithQueryParameter(name, value.ToString());
+            => WithHeaderParameter(name, value.ToString());
+
+        public IMessageRequestBuilder WithQueryParameter(string name, object? value)
+        {
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                throw new ArgumentException($"'{nameof(name)}' cannot be null or whitespace", nameof(name));
+            }
+
+            if (value is null || string.IsNullOrWhiteSpace(value.ToString()))
```

Comment:
If we can remove the `string.IsNullOrWhiteSpace(value.ToString())` and allow empty values, thus supporting empty query values.

```suggestion
            if (value is null)
```